includes/data-stores/class-wc-coupon-data-store-cpt.php

```diff
@@ -69,7 +69,9 @@ class WC_Coupon_Data_Store_CPT extends WC_Data_Store_WP implements WC_Coupon_Dat
 	 * @param WC_Coupon $coupon Coupon object.
 	 */
 	public function create( &$coupon ) {
-		$coupon->set_date_created( time() );
+		if ( ! $coupon->get_date_created( 'edit' ) ) {
+			$coupon->set_date_created( time() );
+		}
```

Comment:
This works, but post status in line 81 should be changed to `future` right depending upon the scheduled date?

Reply:
WordPress will already update the status for us, see:

https://github.com/WordPress/WordPress/blob/55cd633c37bcc602ecf0dc5f3b448f32ca2d8d13/wp-includes/post.php#L3795-L3798

We also does it for products and variations, here an example:

https://github.com/woocommerce/woocommerce/blob/65cdf44adf68799e4dd5280c01083eeb5eeda918/includes/data-stores/class-wc-product-data-store-cpt.php#L109-L111

Reply:
Ah alright! Thanks!